File: CodeForces/contest495/B.py
def solve(a,b):
	ans = 0
	x = 1
	while(True):
		if(x >= a):
			break
		if(a%x == b):
			ans += 1
		x += 1
	return ans

if __name__ == '__main__':
	a,b = map(long, raw_input().split())
	if b >= a:
		print "infinity"
	else:
		print solve(a, b)

File: CodeForces/contest495/B.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main() {
	long a, b;
	cin >> a >> b;
	if(b > a) {
		cout << "0" << endl;
	}
	else if(a == b) {
		cout << "infinity" << endl;
	}
	else {
		long ans = 0;
		long x = 1;
		while(x <= a) {
			if(a%x == b) {
				ans++;
			}
			x++;
		}
		cout << ans << endl;
	}
	return 0;
}
